From 8afd4db75c58cac3e2374fc133fba1bd0232b96d Mon Sep 17 00:00:00 2001
From: Maxime Gasse <maxime.gasse@gmail.com>
Date: Mon, 22 Apr 2019 19:12:37 -0400
Subject: [PATCH 2/4] Vanillafullstrong: bugfixes

---
 src/scip/branch_vanillafullstrong.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/scip/branch_vanillafullstrong.c b/src/scip/branch_vanillafullstrong.c
index eb451e3d80..3f4ac3ee17 100644
--- a/src/scip/branch_vanillafullstrong.c
+++ b/src/scip/branch_vanillafullstrong.c
@@ -150,6 +150,7 @@ SCIP_DECL_BRANCHEXIT(branchExitVanillafullstrong)
    SCIPfreeBufferArrayNull(scip, &branchruledata->cands);
    SCIPfreeBlockMemoryArrayNull(scip, &branchruledata->candscores, branchruledata->ncands);
    branchruledata->ncands = -1;
+   branchruledata->npriocands = -1;
    branchruledata->bestcand = -1;
 
    return SCIP_OKAY;
@@ -184,11 +185,6 @@ SCIP_DECL_BRANCHEXECLP(branchExeclpVanillafullstrong)
    SCIPfreeBufferArrayNull(scip, &branchruledata->cands);
    SCIPfreeBlockMemoryArrayNull(scip, &branchruledata->candscores, branchruledata->ncands);
 
-   /* allocate candidate scores array if needed */
-   if( branchruledata->collectscores ) {
-      SCIP_CALL( SCIPallocClearBlockMemoryArray(scip, &branchruledata->candscores, branchruledata->ncands) );
-   }
-
    /* get branching candidates, either all non-fixed variables or only the
     * fractional ones */
    if( branchruledata->integralcands ) {
@@ -202,6 +198,11 @@ SCIP_DECL_BRANCHEXECLP(branchExeclpVanillafullstrong)
    assert(branchruledata->ncands > 0);
    assert(branchruledata->npriocands > 0);
 
+   /* allocate candidate scores array if needed */
+   if( branchruledata->collectscores ) {
+      SCIP_CALL( SCIPallocClearBlockMemoryArray(scip, &branchruledata->candscores, branchruledata->ncands) );
+   }
+
    SCIP_CALL( runVanillaStrongBranching(scip, branchruledata->cands, branchruledata->ncands, branchruledata->npriocands,
      branchruledata->scoreall, branchruledata->indempotent, branchruledata->candscores,
      &branchruledata->bestcand, &bestdown, &bestup, &bestscore, &bestdownvalid,
@@ -281,6 +282,7 @@ SCIP_RETCODE SCIPincludeBranchruleVanillafullstrong(
    branchruledata->cands = NULL;
    branchruledata->candscores = NULL;
    branchruledata->ncands = -1;
+   branchruledata->npriocands = -1;
    branchruledata->bestcand = -1;
 
    /* include branching rule */
-- 
2.19.1

