From f9050ac8754435e6667d9248ba4233cb9679c48c Mon Sep 17 00:00:00 2001
From: Maxime Gasse <maxime.gasse@gmail.com>
Date: Mon, 3 Jun 2019 17:39:44 -0400
Subject: [PATCH 4/4] Compilation fix

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51dca18a38..fba4013cea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.0)
+add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
 
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_DEBUG} ${CMAKE_C_FLAGS_RELEASE}")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_RELEASE}")
-- 
2.19.1

